storage/tsdb/tsdb.go

```diff
@@ -188,7 +188,8 @@ func Open(path string, l log.Logger, r prometheus.Registerer, opts *Options) (*t
 		}
 	}
 
-	db, err := tsdb.Open(path, l, r, &tsdb.Options{
+	pr := prometheus.WrapRegistererWithPrefix("prometheus_", r)
```

Comment:
We already discussed this, this is an inappropriate use of this feature. Metric names from a library should not change depending on what binary they happen to be in.

Reply:
ok